src/modules/reading-room/utils/map-filters/map-filters.tsx

```diff
@@ -18,78 +21,86 @@ const getSelectLabel = (
 	return options.find((option) => option.value === optionValue)?.label;
 };
 
-export const mapFiltersToTags = (
-	query: Partial<DecodedValueMap<typeof READING_ROOM_QUERY_PARAM_CONFIG>>
-): TagInfo[] => {
-	const searchFilters = (query.search ?? [])
+type ReadingRoomQueryParams = Partial<DecodedValueMap<typeof READING_ROOM_QUERY_PARAM_CONFIG>>;
+
+export const mapSearchToTags = (query: Pick<ReadingRoomQueryParams, 'search'>): TagIdentity[] => {
+	return (query.search || [])
 		.filter((keyword) => !!keyword)
-		.map((keyword) => ({
-			label: (
-				<span>
-					{i18n?.t('modules/reading-room/utils/map-filters/map-filters___trefwoord')}:{' '}
-					<strong>{keyword}</strong>
-				</span>
-			),
-			value: keyword as string,
-			prop: 'search',
-		}));
-
-	const advancedFilters = (query.advanced ?? []).map((advanced: AdvancedFilterQueryValue) => ({
-		label: (
-			<span>
-				{getSelectLabel(METADATA_PROP_OPTIONS(), advanced.prop)}:{' '}
-				<strong>{advanced.val}</strong>
-			</span>
-		),
-		value: advanced.val,
-		prop: ReadingRoomFilterId.Advanced,
-	}));
-
-	return searchFilters.concat(advancedFilters);
+		.map((keyword) => {
+			const label = i18n?.t('modules/reading-room/utils/map-filters/map-filters___trefwoord');
+
+			return {
+				label: (
+					<span>
+						{`${label}: `}
+						<strong>{keyword}</strong>
+					</span>
+				),
+				value: keyword as string,
+				key: SEARCH_QUERY_KEY,
+			};
+		});
 };
 
-export const mapFiltersToQuery = (
-	id: ReadingRoomFilterId,
-	values: unknown
-): AdvancedFilterQueryValue[] | undefined => {
-	switch (id) {
-		case ReadingRoomFilterId.Advanced: {
-			const filters = (values as AdvancedFilterFormState)?.advanced.filter(
-				(filter) => !!filter.value
-			);
-
-			// Map to smaller props to keep query params in url short
-			return filters.length
-				? filters.map((item) => ({
-						prop: item.metadataProp ?? '',
-						op: item.operator ?? '',
-						val: item.value ?? '',
-				  }))
-				: undefined;
+export const mapAdvancedToTags = (
+	query: Pick<ReadingRoomQueryParams, ReadingRoomFilterId.Advanced>
+): TagIdentity[] => {
+	return (query.advanced || []).map((advanced: AdvancedFilter) => {
+		const prop = advanced.prop as MetadataProp;
+		const op = advanced.op as Operator;
+
+		const split = (advanced.val || '').split(SEPARATOR);
+
+		const label = getSelectLabel(getProperties(), prop);
+		let operator = getSelectLabel(getOperators(prop), advanced.op);
+		let value = advanced.val;
+
+		// Convert certain values to be legible
+
+		switch (prop) {
+			case MetadataProp.CreatedAt:
+			case MetadataProp.PublishedAt:
+				if (op === Operator.Between) {
+					value = `${formatMetadataDate(split[0])} - ${formatMetadataDate(split[1])}`;
+					operator = undefined;
+				} else {
+					value = formatMetadataDate(value);
+				}
+				break;
+
+			case MetadataProp.Duration:
+				if (op === Operator.Between) {
+					value = `${split[0]} - ${split[1]}`;
```

Comment:
if advanced.val is undefined, the split value will be an empty array and this will output: undefined - undefined
is that intended?

Reply:
`formatWithLocale` handles this in a later PR